c_glib/gandiva-glib/native-function.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <arrow-glib/arrow-glib.h>
+
+#include <gandiva-glib/function-signature.h>
+
+G_BEGIN_DECLS
+
+/**
+ * GGandivaResultNullableType:
+ * @GGANDIVA_RESULT_NULL_IF_NULL: This means the result validity is an intersection of
+ *   the validity of the children.
+ * @GGANDIVA_RESULT_NULL_NEVER: This means that the result is always valid.
+ * @GGANDIVA_RESULT_NULL_INTERNAL: This means that the result validity depends on some
+ *   internal logic.
+ *
+ * They are corresponding to `gandiva::ResultNullableType` values.
+ */
+typedef enum {
+  GGANDIVA_RESULT_NULL_IF_NULL,
+  GGANDIVA_RESULT_NULL_NEVER,
+  GGANDIVA_RESULT_NULL_INTERNAL
+} GGandivaResultNullableType;
```

Comment:
We need generate `GType` for `enum` in build time.
See:

https://github.com/apache/arrow/blob/master/c_glib/arrow-glib/Makefile.am#L194-L236

https://github.com/apache/arrow/blob/master/c_glib/arrow-glib/meson.build#L178-L185

Reply:
@kou Is this line duplicated?
https://github.com/apache/arrow/blob/master/c_glib/arrow-glib/Makefile.am#L195

Reply:
Ah, we should remove `libarrow_glib_la_generated_headers` and `_sources`.

BTW, could you create a JIRA issue? If we don't associate pull request with JIRA issue, comments on GitHub aren't recorded into JIRA issue.

Reply:
Oh, I forgot to create JIRA issue!

Reply:
I made it https://issues.apache.org/jira/browse/ARROW-5486